### ISSUE TITLE
Got the error: timeout of 30000ms exceeded

### ISSUE THREAD

**Issue by TimHsu-1** [tronprotocol/tronweb#265]
Describe : 
When polling the event for a while it will get the timeout error.

Have try :
1. Set the API key and polling every 3s to prevent  [rate limit](https://developers.tron.network/reference/rate-limits)
2. Renew tronWeb instance if [isConnected](https://github.com/tronprotocol/tronweb/blob/master/src/index.js#L434) return false

update :
I'm using the nile testnet , host (https://nile.trongrid.io/)
Is this the cause the problem ?

Sample code :

```javascript
const TronWebJs = require('tronweb');

async function checkTronWebConnection(tronWeb) {
    let nodes = await tronWeb.isConnected();
    const connected = !Object.entries(nodes).map(([name, connected]) => {
        if (!connected) {
            console.log(`${name} is not connected`);
        }
        return connected;
    }).includes(false);

    if (connected) {
        return true;
    }

    return false;
}

let tronWeb;

async function getTronWeb(newTronWeb = false) {
    let createTronWeb = false;

    if (newTronWeb || tronWeb === undefined) {
        createTronWeb = true;
    } else if (tronWeb instanceof TronWebJs) {
        if (!await checkTronWebConnection(tronClient)) {
            createTronWeb = true;
        }
    }

    if (createTronWeb) {
        const client = new TronWebJs({
            // options,
        });

        tronWeb = client;
    }

    return tronWeb;
}

function watchContractEvents(contractAddress, options) {
    return __awaiter(this, void 0, void 0, function* () {
        while (true) {
            try {
                yield new Promise((resolve, rejects) => __awaiter(this, void 0, void 0, function* () {
                   setInterval(async () => {
                        try {
                            let tronWeb = await getTronWeb();
                            let events = [];
                            let more_events = await tronWeb.getEventResult(contractAddress, {
                                  // options
                            });

                            while (more_events.length && more_events[more_events.length - 1].fingerprint) {
                                events = events.concat(more_events);

                                more_events = await tronClientWs.getEventResult(contractAddress, {
                                    // options
                                });
                            }

                            // handle events
                        } catch (err) {
                            rejects(err);
                        }
                    }, 3000);
                }));
            } catch (err) {
                console.log(err);
            }
        }
    })
}
```

Please help thx a lot.

**Comment by llwslc** [comment]
Please try on the mainnet. (https://api.trongrid.io/)

**Comment by llwslc** [comment]
I just requested 1000 times on both the test network and the main network to fetch events, no timeout issue, please check your network.

**Comment by TimHsu-1** [comment]
I using the API [getEvents](https://developers.tron.network/v4.0/reference/events-by-contract-address) and recored the  response time,  it takes range 1s ~ 16s or timeout happened.

Could I ask does any restrict on the testnet ?
Like if we request too frequently then IP address will be limited ?

 I can't find any document except the [rate limit](https://developers.tron.network/reference/rate-limits).
But according to the document if  I got limited the response should be  
```
{
     "Success": false,
     "Error": "The key exceeds the frequency limit(15), and the query server is suspended for 30s",
     "StatusCode": 403
}
```

Now I changed the IP address to observe the follow-up situation.
